cmd/contrib/main.go

```diff
@@ -180,19 +180,27 @@ func newPlugin() error {
 		}
 	}
 
-	// As a placeholder, assume the field name is the short version (max 7 chars) of the credential name, starting
-	// from the last word. For example:
+	// As a placeholder, assume the field name is the short version (max 7 chars) of the credential name, starting from the last word.
+	//
+	// When the last word of the credential name is greater than seven characters, the last word is used as the field name.
+	//
+	// For example:
 	// "Personal Access Token" => "Token"
 	// "Secret Key" => "Key"
 	// "API Key" => "API Key"
-	var fieldNameSplit []string
+	// "GitHub API Key" => "API Key"
+	// "Credentials" => "Credentials"
 	lengthCutoff := 7
-	for i := range credNameSplit {
+	fieldNameSplit := []string{credNameSplit[len(credNameSplit)-1]}
+	for i := range credNameSplit[:len(credNameSplit)-1] {
+		// Skip credNameSplit's last word because it's already added to fieldNameSplit
+		if i == 0 {
+			continue
+		}
 		word := credNameSplit[len(credNameSplit)-1-i]
 		if len(strings.Join(append(fieldNameSplit, word), " ")) > lengthCutoff {
```

Comment:
A suggestion to address that the `API Key` case is no longer covered: we could also use the original code and add an extra case to this condition:

```suggestion
		if len(strings.Join(append(fieldNameSplit, word), " ")) > lengthCutoff && len(fieldNameSplit) != 0 {
```